### ISSUE TITLE
Inconsistent container build behavior

### ISSUE THREAD

**Issue by kchaung-lilly** [seqeralabs/wave#592]
I would like to use Wave containers for custom multi-tool containers. So far, it's been working great so far and is very handy. However, I've noticed that for certain builds, there are some inconsistent behaviors regarding the `--freeze` option.

Generally, I would like to build Wave containers with the `--freeze` param. However, for some builds, I cannot pull the provided image.  Once I build the container without the `-freeze` param, it provides a different URI that can be pulled. Subsequently, I can then pull the original URI.

Could you please advise? Thank you!

Get the Wave container URI with freeze
```
 $ wave \
        --freeze \
        --conda-package bioconda::vcf2maf=1.6.21 \
        --conda-package bioconda::ensembl-vep=109.3 \
        --conda-package bioconda::bcftools=1.17 
community.wave.seqera.io/library/vcf2maf_ensembl-vep_bcftools:1f8757d3b4b080ba
```

Try to pull the Wave container URI with freeze -- error
```
$ docker pull community.wave.seqera.io/library/vcf2maf_ensembl-vep_bcftools:1f8757d3b4b080ba
Error response from daemon: manifest for community.wave.seqera.io/library/vcf2maf_ensembl-vep_bcftools:1f8757d3b4b080ba not found: manifest unknown: manifest unknown
```

Get the Wave container URI without freeze
```
$ wave \
    --conda-package bioconda::vcf2maf=1.6.21 \
    --conda-package bioconda::ensembl-vep=109.3 \
    --conda-package bioconda::bcftools=1.17 
wave.seqera.io/wt/a5bc4064696d/wave/build:vcf2maf-1.6.21_ensembl-vep-109.3_bcftools-1.17--0991d2cf8ea1ad1f
```

Try to pull the Wave container URI without freeze -- works
```
$ docker pull wave.seqera.io/wt/a5bc4064696d/wave/build:vcf2maf-1.6.21_ensembl-vep-109.3_bcftools-1.17--0991d2cf8ea1ad1f
vcf2maf-1.6.21_ensembl-vep-109.3_bcftools-1.17--0991d2cf8ea1ad1f: Pulling from wt/a5bc4064696d/wave/build
6360b3717211: Already exists 
2ec3f7ad9b3c: Already exists 
7716ca300600: Already exists 
4f4fb700ef54: Already exists 
7834e8feb904: Already exists 
5ac55ff04773: Already exists 
77c7a930b7cc: Already exists 
c864db06f68b: Already exists 
f628b9cff8d1: Already exists 
ba94160d36b7: Already exists 
6f63df1cb8dd: Already exists 
f789ba561933: Pull complete 
dda8f29b496d: Pull complete 
Digest: sha256:a5fc320177428ccb52fbcc97b49d7db9e105c468350948a1dd6f5aa58dcf06bc
Status: Downloaded newer image for wave.seqera.io/wt/a5bc4064696d/wave/build:vcf2maf-1.6.21_ensembl-vep-109.3_bcftools-1.17--0991d2cf8ea1ad1f
wave.seqera.io/wt/a5bc4064696d/wave/build:vcf2maf-1.6.21_ensembl-vep-109.3_bcftools-1.17--0991d2cf8ea1ad1f
```

Now, try to pull the original Wave container URI with freeze -- works
```
$ wave         --conda-package bioconda::vcf2maf=1.6.21         --conda-package bioconda::ensembl-vep=109.3         --conda-package bioconda::bcftools=1.17 --freeze
community.wave.seqera.io/library/vcf2maf_ensembl-vep_bcftools:1f8757d3b4b080ba
```

**Comment by pditommaso** [comment]
TLDR; you should use `--await` option along with `--freeze` to make sure the build is complete before pulling it.

**Comment by kchaung-lilly** [comment]
Thanks, that works well. Will close this issue.

I think it would be helpful to have the `--await` parameter highlighted in front-facing documentation, or maybe a more informative error message than `Error response from daemon: ...`. These larger/longer builds are probably a common use case and the solution is not immediately obvious.

Thank you!
